v3/src/components/graph/adornments/adornments-store-utils.ts

```diff
@@ -0,0 +1,154 @@
+import {getAdornmentContentInfo, getAdornmentTypes, IAdornmentContentInfo} from "./adornment-content-info"
+import {getAdornmentComponentInfo, IAdornmentComponentInfo} from "./adornment-component-info"
+import {PlotTypes} from "./adornment-types"
+import {IMeasure, measures, RulerStateKey} from "./adornment-ui-types"
+import {kMovableLineType} from "./movable-line/movable-line-adornment-types"
+import {kLSRLType} from "./lsrl/lsrl-adornment-types"
+import {IAdornmentsBaseStore} from "./adornments-base-store"
+
+export interface IMeasureMenuItem {
+  checked: boolean
+  componentContentInfo?: IAdornmentContentInfo
+  componentInfo?: IAdornmentComponentInfo
+  disabled?: boolean
+  title: string
+  type: string
+  clickHandler?: () => void
+}
+
+export function isMeasureMenuItem(item: IMeasureMenuItem | IGroupItem): item is IMeasureMenuItem {
+  return item.type !== 'Group'
+}
+
+export interface IGroupItem {
+  title: string
+  type: string
+  rulerStateKey: RulerStateKey | undefined
```

Comment:
```suggestion
  rulerStateKey?: RulerStateKey
```
would be more conventional.